cou/utils/app_utils.py

```diff
@@ -91,6 +91,22 @@ def validate_ovn_support(version: str) -> None:
         )
 
 
+def check_ovn_version_pinning(name: str, version_pinning_config: bool) -> None:
```

Comment:
not-blocking: Why not using this as part of apps itself?

```python
def _check_ovn_version_pinning(self):
    if self.config["enable-version-pinning"].get("value"):
        raise ApplicationError(f"'{self.name}' should set 'enable-version-pinning' before upgrading")
```

Reply:
It's because we would have duplicate code from the principal ovn charm (ovn-central, ovn-dedicated-chassis) and the subordinate (ovn-chassis). That way we have a single place to share the same code

Reply:
They are inheriting from the same place at the end, or we can define mixing.
IMO: having same 2 lines in two places would be better than having one.
Now we have:
```python
check_ovn_version_pinning(self.name, self.config["enable-version-pinning"].get("value"))
super().upgrade_plan_sanity_checks(target, units)
```
and we can have
```python
super().upgrade_plan_sanity_checks(target, units)
if self.config["enable-version-pinning"].get("value"):
    raise ApplicationError(f"'{self.name}' should set 'enable-version-pinning' before upgrading")
```